fastlane/fastlane
# [match] Enable importing developer ID certificates

### Checklist

### Motivation and Context
Resolves #16240

This update allows developer ID certificates to be imported via `fastlane match import`.

### Description
I'm new to the fastlane codebase, but it appears that match tries to be helpful by verifying the certificates you are trying to import are available in the developer portal. It pulls down the certificates and then filters them by the type specified.

Currently, the only certificate types match import will filter with are `development`, `distribution`, and `enterprise` even though match supports `developer_id` certificates ([source](https://github.com/fastlane/fastlane/blob/6527088a3f1daa16fae6a56e18833cac4d757512/match/lib/match/module.rb#L26)).

This change adds `developer_id` as a supported type that import can use to filter results from the developer portal.

### Testing Steps
To see the error that prompted this PR you can apply this patch and then run `bundle exec rspec match`. *The patch just reverts the change to `import.rb` to the current state.*

[original.patch.zip](https://github.com/fastlane/fastlane/files/4389548/original.patch.zip)

I have added a new test that should provide coverage for this use case. You can verify this test passes by running `bundle exec rspec match`.
